internal/pubsub/acks_test.go

```diff
@@ -177,10 +178,11 @@ func TestTooManyAcksForASingleBatchGoIntoMultipleBatches(t *testing.T) {
 func TestAckDoesNotBlock(t *testing.T) {
 	ctx := context.Background()
 	m := &driver.Message{}
+	delay := time.Second
 	ds := &ackingDriverSub{
 		q: []*driver.Message{m},
 		sendAcks: func(_ context.Context, ackIDs []driver.AckID) error {
-			select {}
+			time.Sleep(delay)
```

Comment:
I would but it hangs the way you suggested. That's because the Subscription constructor makes its own context, as suggested earlier by @jba.

In order to fix this, I'll have to change the Subscription constructor to take a context argument again, and then update all its uses to pass in a context. I think it's worth doing but beyond the scope of this CL. Filed an issue: https://github.com/google/go-cloud/issues/815.

Reply:
I added some comments to that issue.

Couldn't you do this with a WaitGroup instead of a sleep?